src/Quizzer.API.BL/Facades/QuestionFacade.cs

```diff
@@ -217,6 +215,22 @@ public class QuestionFacade : FacadeBase<QuestionEntity, QuestionDetailModel> {
                     answer.Order = order;
                     selectedAnswerRepository.Update(answer);
                 }
+
+                using IRepository<AnswerEntity> answerRepository = uow.GetRepository<AnswerEntity>();
+
+                AnswerEntity[] orderredAnswers = answerRepository.GetByPredicate(a => a.QuestionId == answerEntity.QuestionId)
+                    .OrderBy(a => a.Order).ToArray();
+                SelectedAnswerEntity[] orderredUserAnswers = userAnswer.SelectedAnswers.OrderBy(a => a.Order).ToArray();
+
+                userAnswer.AnswerScore = 0;
+                for (int i = 0; i < orderredAnswers.Length; i++) {
+                    AnswerEntity currentAnswer = orderredAnswers[i];
+                    var index = Math.Min(orderredUserAnswers.Length - 1, orderredAnswers.Length);
+                    SelectedAnswerEntity currentUserAnswer = orderredUserAnswers[index];
+
```

Comment:
When there is `var index = Math.Min(orderredUserAnswers.Length - 1, orderredAnswers.Length);` then the `currentUserAnswer` will always be the same, right? But why? Did you mean sth like this?
<pre>var index = Math.Min(orderredUserAnswers.Length - 1, <b>i</b>);</pre>

Reply:
Yes that is correct, fixed. Was doing that when I was exhausted 😄 ...